text/3486-feature-deprecation.md

```diff
@@ -0,0 +1,159 @@
+- Feature Name: feature-deprecation
+- Start Date: 2023-09-09
+- RFC PR: [rust-lang/rfcs#3486](https://github.com/rust-lang/rfcs/pull/3486)
+- Rust Issue:
+  [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+
+[summary]: #summary
+
+This RFC describes a new key under `features` in `Cargo.toml` to indicate that a
+feature is deprecated.
+
+Please see the parent meta RFC for background information: [`feature-metadata`].
+
+# Motivation
+
+[motivation]: #motivation
+
+Cargo features are widely used and typically have lifecycles the same as other
+API components. There is not currently a way to indicate that a feature is
+intended for removal and warn about it: This RFC proposes a `deprecated` key
+that shows this information.
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+A new `deprecated` key will be allowed for features, defaulting to `false` if
+not specified. If specified, the value can be either a boolean, a string, or an
+object with `since` and/or `note` keys. Cargo will warn downstream crates using
+this feature.
+
+```toml
+[features]
+foo = { enables = [], deprecated = true }
+foo = { enables = [], deprecated = "this works as a note" }
+bar = { enables = [], deprecated = { since = "1.2.3", note = "don't use this!" } }
+```
+
+See [`feature-metadata`] for information about `enables`.
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+`deprecated` should be thought of as the equivalent of the [`deprecated`]
+attribute in Rust source. The value can be a boolean, string, or an object with
+`since` or `note` keys. Schema rules are as follows:
+
+- If a boolean value, `false` indicates not deprecated and `true` indicates
```

Comment:
It's not clear to me what the benefit is of having a boolean form. `deprecated=true` provides little information to the user, and it seems relatively little effort to at least provide a string explaining why it is deprecated or what to use instead.

Reply:
I assume its to be consistent with the attribute but I see little lost in being inconsistent on this part.